### PR TITLE
Add types for try-catch

### PR DESCRIPTION
**Pull request by coderaiser**
Added types for [try-catch](https://www.npmjs.com/package/try-catch) package

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/try-catch/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "strictNullChecks": true,
+        "strictFunctionTypes": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true
```
Comment (peterblazejewicz) [inline types/try-catch/tsconfig.json:18]:
```suggestion
        "forceConsistentCasingInFileNames": true
```

File: types/try-catch/try-catch-tests.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import tryCatch from "try-catch";
```
Comment (peterblazejewicz) [inline types/try-catch/try-catch-tests.ts:1]:
```suggestion
import tryCatch = require("try-catch");
```

File: types/try-catch/tslint.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "rules": {
+        "ban-types": false,
+        "npm-naming": [true, {
+            "mode": "code",
+            "errors":[
+                ["NeedsExportEquals", false]
+            ]
+        }]
+    }
+}
```
Comment (peterblazejewicz) [inline types/try-catch/tslint.json:12]:
```suggestion
    "extends": "dtslint/dt.json"
}
```
why you needed those? You will need inline exclusion for `Function`, which is OK, just add it inline

Comment (coderaiser) [reply types/try-catch/tslint.json]: I'm goin to use `try-catch` with `imports` only, so would be great not to add `esModuleInterop` to every `tsconfig.json`. But as I see most packages uses this variant, so I'll need it anyway.

File: types/try-catch/tslint.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+    "extends": "dtslint/dt.json"
+}
+
```
Comment (peterblazejewicz) [inline types/try-catch/tslint.json:4]:
```suggestion
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@coderaiser thx!

**Comment by coderaiser** [comment]
Ready to merge

**Pull request merged**